lepton/utils.go

```diff
@@ -14,3 +15,58 @@ func exitWithError(errs string) {
 func arrayToString(a []int, delim string) string {
 	return strings.Trim(strings.Replace(fmt.Sprint(a), " ", delim, -1), "[]")
 }
+
+// isDomainValid returns an error if the domain name is not valid
+// See https://tools.ietf.org/html/rfc1034#section-3.5 and
+// https://tools.ietf.org/html/rfc1123#section-2.
+// code extracted from https://gist.github.com/chmike/d4126a3247a6d9a70922fc0e8b4f4013
+func isDomainValid(name string) error {
```

Comment:
this is fine to have - just curious why you ran into this?

Reply:
The function seemed pretty complete to grant domain names have the right format and prevent errors on creating the rules in a cloud provider.